pennylane_qiskit/converter.py

```diff
@@ -328,56 +327,62 @@ def _function(*args, params: dict = None, wires: list = None, **kwargs):
-        mid_circ_meas, terminal_meas = [], []
+        terminal_meas = []
+        mid_circ_meas, mid_circ_regs = [], {}
 
         # Processing the dictionary of parameters passed
-        for idx, (op, qargs, _) in enumerate(qc.data):
-            # the new Singleton classes have different names than the objects they represent, but base_class.__name__ still matches
-            instruction_name = getattr(op, "base_class", op.__class__).__name__
-
-            operation_wires = [wire_map[hash(qubit)] for qubit in qargs]
-
+        for idx, (ops, qargs, cargs) in enumerate(qc.data):
+            # the new Singleton classes have different names than the objects they represent,
+            # but base_class.__name__ still matches
+            instruction_name = getattr(ops, "base_class", ops.__class__).__name__
             # New Qiskit gates that are not natively supported by PL (identical
             # gates exist with a different name)
             # TODO: remove the following when gates have been renamed in PennyLane
             instruction_name = "U3Gate" if instruction_name == "UGate" else instruction_name
 
-            # pylint:disable=protected-access
-            if (
-                instruction_name in inv_map
-                and inv_map[instruction_name] in pennylane_ops._qubit__ops__
-            ):
-                # Extract the bound parameters from the operation. If the bound parameters are a
-                # Qiskit ParameterExpression, then replace it with the corresponding PennyLane
-                # variable from the unbound_params dictionary.
-
-                pl_parameters = []
-                for p in op.params:
-                    _check_parameter_bound(p, unbound_params)
-
-                    if isinstance(p, ParameterExpression):
-                        if p.parameters:  # non-empty set = has unbound parameters
-                            ordered_params = tuple(p.parameters)
-
-                            f = lambdify(ordered_params, p._symbol_expr, modules=qml.numpy)
-                            f_args = []
-                            for i_ordered_params in ordered_params:
-                                f_args.append(unbound_params.get(i_ordered_params))
-                            pl_parameters.append(f(*f_args))
-                        else:  # needed for qiskit<0.43.1
-                            pl_parameters.append(float(p))  # pragma: no cover
-                    else:
-                        pl_parameters.append(p)
-
-                execute_supported_operation(
-                    inv_map[instruction_name], pl_parameters, operation_wires
-                )
+            # Define operator builders and helpers
+            operation_func = None
 
-            elif instruction_name in dagger_map:
-                gate = dagger_map[instruction_name]
-                qml.adjoint(gate)(wires=operation_wires)
+            def operation_overlapper(op):
+                return op
```

Comment:
Actually - it looks like since it didn't work out for this to also include `conds`, this is only ever either "do nothing" or `adjoint`. Then when it's finally applied, it's applied as `operation_overlapper(operation_func)(*operation_args, **operation_kwargs)`. 

Can't we just apply `adjoint` to `operation_func` the relevant spot, i.e. if line 374 is:

```
if instruction_name in dagger_map:
    operation_func = qml.adjoint(dagger_map[instruction_name])
```

and then we don't need the `operation_overlapper` at all?

Reply:
I thought so, too! Removed the `overlapper` based on this.